api/openstorage.yaml

```diff
@@ -901,48 +923,107 @@ paths:
-    delete:
-      summary: Delete Volume
-      description: |
-        Delete Volume
-      parameters:
-        - name: uuid
-          in: path
-          description: Volume UUID
-          required: true
-          type: string
-          format: uuid
-      tags:
-        - Volumes
-      responses:
-        '200':
-          description: Volume Deleted
-          schema:
-            $ref: '#/definitions/UUID'
+      put:
+        summary: Change volume state
+        description: |
+          Attach, detach, format, mount and unmount operations
+        parameters:
+          - name: uuid
+            in: path
+            description: |
+             Volume UUID as path parameter example:
+             PUT /v1/osd-volumes/myvolume
+            required: true
+            type: string
+            format: uuid
+          - name: state
+            in: body
+            description: |
+              Desired Volume State attach example:
+               ``` json
+                   {
+                     "action": {
+                        "attach: 2"
+                        }
+                   }
+              ```
+              Desired Volume State detach example:
+               ``` json
+                   {
+                     "action": {
+                        "attach: 1"
+                        }
+                   }
+              ```
+              Desired Volume State mount example:
+               ``` json
+                   {
+                     "action": {
+                        "mount: 2",
+                        "mount_path:tmp"
+                        }
+                   }
+              ```
+              Desired Volume State unmount example:
+               ``` json
+                   {
+                     "action": {
+                        "mount: 1",
+                        "mount_path:tmp"
+                        }
+                   }
+              ```
+            required: true
+            schema:
+              $ref: '#/definitions/VolumeStateParams'
+        tags:
+          - Volumes
+        responses:
+          '200':
+            description: |
+             Operation result of Attach/Detach/Mount/Unmount
+            schema:
+              $ref: '#/definitions/VolumeStateParams'
+            examples:
+             application/json: |-
+              {
+                 "cmd": "attach",
+                 "status": "Ok",
+                 "desc": "Volume successfully attached at: /dev/pxd/pxd186672328357254033",
+                 "uuid": [
+                  "h"
+                 ]
+              }
+      delete:
+        summary: Delete Volume
+        description: |
+          Delete Volume
+        parameters:
+          - name: uuid
+            in: path
+            description: |
+             Volume UUID as path parameter example:
+             DELETE /v1/osd-volumes/myvolume
+            required: true
+            type: string
+            format: uuid
```

Comment:
What does "type:string' and "format:uuid" mean? We allow both volume name and volume uuid to be specified.